addon-tools/compare-to-helm/README.md

```diff
@@ -0,0 +1,148 @@
+This can be complicated for references including multiple templates, because it requires going over all the templates,
+and gathering a list of which values are required to be able to build the CR.
+
+To address this problem the Tool creates a values.yaml file structure that contains values that are required in order
+to build the CRs. Then the user only needs to fill in the values.yaml file with the default values without needing to
+think about which values are needed and without needing to build the structure.
+The list of values that are needed are collected by a traversal on the static form of template.
+Due to the big amount of possibles in go templating syntax some cases where values are used can be not detected.
+This version covers all the basic use-cases, More complex cases can be added by demand. Additionally, different
+approaches to this issue can be added.  
+The initial implementation of the values gathering is a great place to start with, and for complex cases users may will
+need to so some minor modifications. This i also a one time thing per reference, for references that have already been converted
+the previous values.yaml can be used as a base for the values.yaml file. Most of the work is in the initial conversion.
+
+example generated values.yaml structure (does not include the default values):
+
+```yaml
+#gnerated from: pkg/compare/testdata/AllRequiredTemplatesExistAndThereAreNoDiffs/reference/metadata.yaml
+cm:
+- {} # Template with no user defined content  
+cr:
+- {}
+crb:
+- {}
+deploymentDashboard:
+- {}
+deploymentMetrics:
+- spec:
+    template:
+      spec: {}
+ns:
+- {}
+rb:
+- {}
+role:
+- apiVersion: {}
+  metadata:
+    name: {}  # user needs to replace {} with default value (for many {} in example)
+    namespace: {}
+sa:
+- apiVersion: {}
+  metadata:
+    name: {}
+secret:
+- data: {}
+  metadata:
+    name: {}
+service:
+- metadata:
+    labels: {}
+    name: {}
+  spec:
+    ports:
+      "0":
+        port: {}
+        targetPort: {}
+    selector:
+      k8s-app: {}
+```
+
+## Auto Extracting of default values from Existing CRs
+
+another feature that can help in initial building of values.yaml files is extracting default values from existing CRs,
+The command gets a path to a directory that should include CRs that there name is the same as the corresponding template.
+For each template that includes a CR with the same name in the defaults directory the tool will extract the defaults values
+from the corresponding CR.
+
+## Run
+
+Basic usage:
+
+```shell
+# create helm chart
+helm-convert -r ./reference/metadata.yaml -n ChartDir 
+# Render templates, resulting in creation of default CRs that follow the reference 
+helm template renderedRef ChartDir/ChartDir/ --output-dir renderedRefDir
+```
+
+Auto Extracting of default values from Existing CRs:
+
+```shell
+helm-convert -r ./reference/metadata.yaml -n ChartDir -d ExstisingCRsDir
```

Comment:
nothing, at least as i implemented it because it may contain CRs that are not supposed to be in the reference or different unrelated yaml files .
Is there any behavior you would of expect?

Reply:
I would expect it to be added to as helm chart. That way you can keep it as part of the cycle.

Reply:
I think there are a couple of interesting use cases here. We should articulate a set of user stories to cover this functionality to help guide how it is implemented. eg
- Pulling defaults from a directory to populate the values.yaml for an existing reference. May want to highlight crs with no existing reference for consideration to pull them in as new reference CRs.
- Pulling defaults from a live cluster
- etc.